Dockerfile

```diff
@@ -54,20 +90,36 @@ RUN go get -v ./...
 RUN mkdir caddy && cd caddy
 COPY go.mod go.sum ./
 
-RUN go get -v ./... && \
-    cd ..
+RUN go get -v ./...
 
 COPY . .
 
+# todo: automate this?
+# see https://github.com/docker-library/php/blob/master/8.2-rc/bullseye/zts/Dockerfile#L57-L59 for php values
+ENV CGO_LDFLAGS="-lssl -lcrypto -lreadline -largon2 -lcurl -lonig -lz $PHP_LDFLAGS" CGO_CFLAGS=$PHP_CFLAGS CGO_CPPFLAGS=$PHP_CPPFLAGS
+
 RUN cd caddy/frankenphp && \
     go build && \
     cp frankenphp /usr/local/bin && \
-    cp /go/src/app/caddy/frankenphp/Caddyfile /etc/Caddyfile && \
-    rm -Rf /go
+    cp /go/src/app/caddy/frankenphp/Caddyfile /etc/Caddyfile
+
+FROM php:8.2.0RC4-zts-bullseye AS final
+
+COPY --from=mlocati/php-extension-installer /usr/bin/install-php-extensions /usr/local/bin/
 
 WORKDIR /app
 
 RUN mkdir -p /app/public
 RUN echo '<?php phpinfo();' > /app/public/index.php
 
-CMD [ "frankenphp", "run", "--config", "/etc/Caddyfile" ]
+COPY --from=builder /usr/local/bin/frankenphp /usr/local/bin/frankenphp
+COPY --from=builder /etc/Caddyfile /etc/Caddyfile
+
+COPY --from=php-base /usr/local/include/php/ /usr/local/include/php
+COPY --from=php-base /usr/local/lib/libphp.* /usr/local/lib
+COPY --from=php-base /usr/local/lib/php/ /usr/local/lib/php
+COPY --from=php-base /usr/local/php/ /usr/local/php
+COPY --from=php-base /usr/local/bin/ /usr/local/bin
+COPY --from=php-base /usr/src /usr/src
```

Comment:
This is mostly an optimization. It keeps the Docker Image quite small. Alternatively, we can base on the `php-base` image, but then the image is ~2gb for some reason.

Reply:
It's safer and cleaner to do that imho